src/software/backend/input/network/ssl_protobuf_reader.cpp

```diff
@@ -1,43 +1,36 @@
-#include "software/backend/input/network/network_filter.h"
-
-#include "shared/constants.h"
-#include "software/constants.h"
-#include "software/parameter/dynamic_parameters.h"
-#include "software/proto/messages_robocup_ssl_detection.pb.h"
-#include "software/proto/messages_robocup_ssl_geometry.pb.h"
-#include "software/sensor_fusion/refbox_data.h"
+#include "software/backend/input/network/ssl_protobuf_reader.h"
 
 // We can initialize the field_state with all zeroes here because this state will never
 // be accessed by an external observer to this class. the getFieldData must be called to
 // get any field data which will update the state with the given protobuf data
-NetworkFilter::NetworkFilter()
-    : field_state(0, 0, 0, 0, 0, 0, 0, Timestamp::fromSeconds(0)),
-      ball_state(Point(), Vector(), Timestamp::fromSeconds(0)),
-      friendly_team_state(Duration::fromMilliseconds(
-          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
-      enemy_team_state(Duration::fromMilliseconds(
-          Util::Constants::ROBOT_DEBOUNCE_DURATION_MILLISECONDS)),
-      ball_filter(BallFilter::DEFAULT_MIN_BUFFER_SIZE,
-                  BallFilter::DEFAULT_MAX_BUFFER_SIZE),
-      friendly_team_filter(),
-      enemy_team_filter()
+SSLProtobufReader::SSLProtobufReader()
+    : field_state(0, 0, 0, 0, 0, 0, 0, Timestamp::fromSeconds(0))
```

Comment:
Why is field_state the only thing still needed?

Reply:
we use it to return the previous detected field in case createFieldFromPacketGeometry cannot create a field due to a missing optional part of the message